event_sync/images.go

```diff
@@ -0,0 +1,182 @@
+package event_sync
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io"
+	"mime/multipart"
+	"net/http"
+	"path"
+	"strconv"
+	"strings"
+)
+
+type Image struct {
+	Buffer []byte
+	Name   string
+}
+
+type EventbriteUploadData struct {
+	UploadData struct {
+		AWSAccessKeyID string `json:"AWSAccessKeyId"`
+		Bucket         string `json:"bucket"`
+		ACL            string `json:"acl"`
+		Key            string `json:"key"`
+		Signature      string `json:"signature"`
+		Policy         string `json:"policy"`
+	} `json:"upload_data"`
+	UploadURL   string `json:"upload_url"`
+	UploadToken string `json:"upload_token"`
+}
+
+func downloadImageFromFacebook(imageUrl string) (Image, error) {
+	var image Image
+
+	resp, err := http.Get(imageUrl)
+	if err != nil {
+		return image, err
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return image, errors.New("failed to get image from Facebook. Status: " + strconv.Itoa(resp.StatusCode))
+	}
+
+	image.Buffer = make([]byte, resp.ContentLength)
+	_, err = io.ReadFull(resp.Body, image.Buffer)
```

Comment:
FYI, you could also do `image.Buffer, err = ioutil.ReadAll(resp.Body)` if you wanted the code to be a little simpler, but this code is actually fine and more efficient, because it knows the size of the buffer in advance so it only allocates once.

Reply:
I'll use that code instead. I spent a fair amount of time trying to figure out how to get this to work, so your simpler option seems better since the efficiency here doesn't matter much.